File: graphics-next/src/app/modifiers.ts
Next.js היא ספריית JavaScript שמבוססת על React, ולכן המרה של קוד Python ל-Next.js תכלול שינוי משמעותי בסינטקס ובגישה. כמו כן, Next.js אינה מכילה מודלים ומתודות מובנות לניהול סל קניות, כך שנצטרך להשתמש בספרייה חיצונית או לבנות את הפונקציונליות מאפס.

להלן דוגמה לקובץ JavaScript שמממש את הפונקציונליות של המחלקות LimitBasketModifier ו-FilterBasketModifier באמצעות React ו-Next.js:

```javascript
import { useState } from 'react';

function Basket() {
  const [items, setItems] = useState([]);

  const limitItemQuantity = (item) => {
    if (item.quantity > 1) {
      item.quantity = 1;
    }
  };

  const filterBasketItems = (basket, isReplenishment, isPurchasingSubscription) => {
    const newItems = basket.items.filter((item) => {
      if (isReplenishment && item.code[0] !== 'B') {
        return false;
      }

      if (isPurchasingSubscription && item.code[0] !== 'S') {
        return false;
      }

      if (item.code[0] in ['B', 'S']) {
        return false;
      }

      return true;
    });

    setItems(newItems);
  };

  // ... render the basket
}

export default Basket;
```

שים לב שהקוד הזה הוא רק דוגמה ואינו מכיל את כל הפונקציונליות של הקוד המקורי. לדוגמה, הוא לא משנה את מספר הפריטים בסל באופן דינאמי בהתאם לבקשות המשתמש, והוא לא משמר את השינויים במסד הנתונים.

File: app/shop/modifiers.py
from django.http import HttpRequest  # type: ignore
from salesman.basket.modifiers import BasketModifier  # type: ignore

from shop.models import (
    Basket,
    BasketItem,
)


class LimitBasketModifier(BasketModifier):
    identifier = 'limit'

    def setup_item(self, item: BasketItem, request: HttpRequest) -> None:
        if item.quantity > 1:
            item.quantity = 1
            item.save()


class FilterBasketModifier(BasketModifier):
    identifier = 'filter'

    def process_basket(self, basket: Basket, request: HttpRequest) -> None:
        is_replenishment = basket.extra.pop('is_replenishment', False)
        is_purchasing_subscription = basket.extra.pop('is_purchasing_subscription', False)

        for basket_item in basket.items.all():

            if is_replenishment and basket_item.code[0] != 'B':
                basket_item.delete()

            elif is_purchasing_subscription and basket_item.code[0] != 'S':
                basket_item.delete()

            elif basket_item.code[0] in {'B', 'S'}:
                basket_item.delete()

        basket.save()
